zschema/compounds.py

```diff
@@ -192,8 +192,11 @@ def validate(self, value):
             raise DataValidationException("record is not a dict", str(value))
         for subkey, subvalue in value.items():
             if subkey not in self.definition:
-                raise DataValidationException("%s is not a valid subkey of root",
-                                              subkey)
+                port = Port(subkey)
```

Comment:
While there might be a bug in that, patching in a hack to cover that up is not a solution.